### DISCUSSION TITLE
what is the use of `trpc.dehydrate()`?

### DISCUSSION THREAD

**Discussion by HananoshikaYomaru** [trpc/trpc#1997] [Q&A]
In [this example](https://trpc.io/docs/ssg), what is the use of `trpc.dehydrate()`? 

```tsx
export default function PostViewPage(
  props: InferGetStaticPropsType<typeof getStaticProps>,
) {
  const { id } = props;
  const postQuery = trpc.useQuery(['post.byId', { id }]);

  if (postQuery.status !== 'success') {
    // won't happen since we're using `fallback: "blocking"`
    return <>Loading...</>;
  }
  const { data } = postQuery;
  return (
    <>
      <h1>{data.title}</h1>
      <em>Created {data.createdAt.toLocaleDateString('en-us')}</em>

      <p>{data.text}</p>

      <h2>Raw data:</h2>
      <pre>{JSON.stringify(data, null, 4)}</pre>
    </>
  );
}
```

Only `id` is deconstructed from props and trpcState doesn't seems to be used. I tried this example, but the `Loading...` text is still coming out even when the comment said it would not. What could be the problem?

**Comment by KATT** [comment]
Create reproduction

**Reply by KATT** [reply]
The dehydrate is to dehydrate the server cache, this particular view probably wouldn't need the `props: InferGetStaticPropsType<typeof getStaticProps>,`-part
